src/useRouteQuery/useRouteQuery.ts

```diff
@@ -1,90 +1,111 @@
-    return query[key]
-  }
-
-  const remove = (key: string): void => {
-    delete query[key]
-  }
-
-  const query: LocationQuery = new Proxy({}, {
-    get(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.get(target, property)
-      }
-
-      const query = applyQueryOperations(route.query, operations)
-
-      return query[property]
-    },
-    set(target, property, value) {
-      if (typeof property === 'symbol' || typeof value !== 'string') {
-        return Reflect.set(target, property, value)
-      }
-
-      const operation = updateOperationFactory(property, value)
-
-      operations.push(operation)
-
-      return true
-    },
-    deleteProperty(target, property) {
-      if (typeof property === 'symbol') {
-        return Reflect.deleteProperty(target, property)
-      }
-
-      const operation = deleteOperationFactory(property)
-
-      operations.push(operation)
-
-      return true
-    },
-  })
-
   const update = (): void => {
     const query = applyQueryOperations(route.query, operations)
 
+    operations.splice(0)
+
     router.push({ query })
   }
 
   watch(operations, () => {
+    if (!operations.length) {
+      return
+    }
+
     clearInterval(interval)
     interval = setTimeout(update)
   })
 
-  return {
-    clear,
-    set,
-    get,
-    remove,
-    query,
+  const useRouteQuery = (): UseRouteQuery => {
+    router = useRouter()
+    route = useRoute()
+
+    const clear = (): void => {
+      router.push({ query: {} })
+    }
+
+    const set = (key: string, value: LocationQueryValue | LocationQueryValue[]): void => {
+      query[key] = value
+    }
+
+    const get = (key: string): LocationQueryValue | LocationQueryValue[] => {
+      return query[key]
+    }
+
+    const remove = (key: string): void => {
+      delete query[key]
+    }
+
+    const query: LocationQuery = new Proxy({}, {
+      get(target, property) {
+        if (typeof property === 'symbol') {
+          return Reflect.get(target, property)
+        }
+
+        const query = applyQueryOperations(route.query, operations)
+
+        return query[property]
+      },
+      set(target, property, value) {
+        if (typeof property === 'symbol' || typeof value !== 'string') {
+          return Reflect.set(target, property, value)
+        }
+
+        const operation = updateOperationFactory(property, value)
+
+        operations.push(operation)
+
+        return true
+      },
+      deleteProperty(target, property) {
+        if (typeof property === 'symbol') {
+          return Reflect.deleteProperty(target, property)
+        }
+
+        const operation = deleteOperationFactory(property)
+
+        operations.push(operation)
+
+        return true
+      },
+      ownKeys() {
+        const query = applyQueryOperations(route.query, operations)
+
+        return Object.keys(query)
+      },
+      has(target, property) {
+        return property in route.query
```

Comment:
why did you choose not to call `applyQueryOperations` here like you do for other `get`-ish functions?

Reply:
Wasn't intentional. And using `applyOperations` here makes sense because you should be able to check for the presence of a key even if the operations haven't been applied. Good catch!